layers/+lang/go/funcs.el

```diff
@@ -37,7 +37,14 @@
 (defun spacemacs//go-setup-backend-lsp ()
   "Setup lsp backend"
   (if (configuration-layer/layer-used-p 'lsp)
-      (lsp)
+      (progn
+        ;; without setting lsp-prefer-flymake to :none
+        ;; golangci-lint errors won't be reported
+        (if go-use-golangci-lint
```

Comment:
nit: you could use `(when ...` and remove the `progn`.

Reply:
👍